### PR TITLE
IPv6 fragmentation: don't leak memory/packets

### PR DESCRIPTION
**Pull request by kbara**

### CODE REVIEW

File: src/apps/lwaftr/lwaftr.lua [hunk 1]
```diff
@@ -531,32 +531,40 @@ local function get_ipv6_dst_ip(lwstate, pkt)
...
+   table.insert(cache[key], frag)
+   return cache[key]
+end
+
+local function clean_ipv6_fragment_cache(lwstate, frags)
+   local key = key_ipv6_frag(lwstate, frags[1])
+   lwstate.fragment6_cache[key] = nil
+   for i=1,#frags do
+      packet.free(frags[i])
+   end
```
Comment (dpino) [inline src/apps/lwaftr/lwaftr.lua]: This @eugeneia comment in a recent PR came to my mind https://github.com/SnabbCo/snabbswitch/pull/664

" Do not keep invalid pointers around. If you free a packet held in a variable that can be accessed outside the lexical scope (for instance an object slot), null it right away."

Maybe it's not necessary here, but what do you think about nulling the packet as well?

```lua
packet.free(frags[i])
frags[i] = nil
```

**Comment by dpino** [comment]
Besides comment, LGTM.

**Comment by kbara** [comment]
Frags is essentially a local in the caller that is never used again after the call to clean the packet cache. We could explicitly nil frags in the caller, but nil-ing elements of it strikes me as a bad idea.

**Comment by dpino** [comment]
Sounds reasonable, LGTM.

**Comment by kbara** [comment]
Merging as per jabber discussion.

**Comment by aperezdc** [comment]
Reatroactively LGTM :wink:

**Pull request merged**